homeassistant/components/tuya/sensor.py

```diff
@@ -579,6 +579,76 @@ class TuyaSensorEntityDescription(SensorEntityDescription):
             subkey="voltage",
         ),
     ),
+    # Robot Vacuum
+    # https://developer.tuya.com/en/docs/iot/fsd?id=K9gf487ck1tlo
+    "sd": (
+        TuyaSensorEntityDescription(
+            key=DPCode.CLEAN_AREA,
+            name="Cleaning Area",
+            icon="mdi:texture-box",
+            state_class=SensorStateClass.MEASUREMENT,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.CLEAN_TIME,
+            name="Cleaning Time",
+            icon="mdi:progress-clock",
+            state_class=SensorStateClass.MEASUREMENT,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.TOTAL_CLEAN_AREA,
+            name="Total Cleaning Area",
+            icon="mdi:texture-box",
+            state_class=SensorStateClass.TOTAL_INCREASING,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.TOTAL_CLEAN_TIME,
+            name="Total Cleaning Time",
+            icon="mdi:history",
+            state_class=SensorStateClass.TOTAL_INCREASING,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.TOTAL_CLEAN_COUNT,
+            name="Total Cleaning Times",
+            icon="mdi:counter",
+            state_class=SensorStateClass.TOTAL_INCREASING,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.DUSTER_CLOTH,
+            name="Duster Cloth Life",
+            icon="mdi:ticket-percent-outline",
+            state_class=SensorStateClass.MEASUREMENT,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.EDGE_BRUSH,
+            name="Side Brush Life",
+            icon="mdi:ticket-percent-outline",
+            state_class=SensorStateClass.MEASUREMENT,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.FILTER_LIFE,
+            name="Filter Life",
+            icon="mdi:ticket-percent-outline",
+            state_class=SensorStateClass.MEASUREMENT,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.ROLL_BRUSH,
+            name="Rolling Brush Life",
+            icon="mdi:ticket-percent-outline",
+            state_class=SensorStateClass.MEASUREMENT,
+        ),
+        TuyaSensorEntityDescription(
+            key=DPCode.FAULT,
```

Comment:
This provides a bitmap, according to the documentation. How is that processed?

Reply:
There is no definition of the bitmap in Tuya's documentation. Soon the sensor will bring the fault code reported by the device.

![image](https://user-images.githubusercontent.com/31328123/144581072-5f3c7c98-5bf1-45d2-9909-73891afb2990.png)

Reply:
If we don't handle bitmap, we should not add it. As this type is unknown, it will always result in `None`. So this currently creates an permanent "Unknown" sensor (regards if the sensors has a value).

Reply:
Is adding as a vacuum entity attribute acceptable?

Reply:
Removed the fault sensor.